openwisp_firmware_upgrader/admin.py

```diff
@@ -179,7 +180,24 @@ def change_view(self, request, object_id, form_url='', extra_context=None):
         extra_context = extra_context or {}
         upgrade_url = f'{app_label}_build_changelist'
         extra_context.update({'upgrade_url': upgrade_url})
-        return super().change_view(request, object_id, form_url, extra_context)
+        # preventing change_view to throw an error
+        try:
+            return super().change_view(request, object_id, form_url, extra_context)
+        except FileNotFoundError as e:
+            path = e.filename
```

Comment:
Can't we just delete the object here, add a message for the user with the message framework and redirect to the admin list page?

Keep in mind this is an exceptional case which happens only if someone messes with the filesystem.